tests/rptest/services/redpanda.py

```diff
@@ -736,7 +736,8 @@ def __init__(self,
                  pandaproxy_config: Optional[PandaproxyConfig] = None,
                  schema_registry_config: Optional[SchemaRegistryConfig] = None,
                  disable_cloud_storage_diagnostics=False):
-        super(RedpandaService, self).__init__(context, num_nodes=num_brokers)
+        super(RedpandaServiceBase, self).__init__(context,
```

Comment:
I think you'd need to separate out this constructor into a base part and a RedpandaService part: e.g. `environment` is probably not going to be usable on kubernetes where we can set redpanda's environment variables, `superuser` may also not be settable, etc.

Reply:
i've reduced the parameters in the constructor of `RedpandaServiceBase` to only include what is used in the base class instead of taking the entire signature from the original `RedpandaService` class.